packages/flutter_tools/lib/src/build_system/filestore.dart

```diff
@@ -153,24 +158,47 @@ class FileHashStore {
 
   /// Computes a hash of the provided files and returns a list of entities
   /// that were dirty.
-  Future<List<File>> hashFiles(List<File> files) async {
+  Future<List<File>> updateFiles(List<File> files) async {
     final List<File> dirty = <File>[];
-    final Pool openFiles = Pool(kMaxOpenFiles);
-    await Future.wait(<Future<void>>[
-       for (File file in files) _hashFile(file, dirty, openFiles)
-    ]);
+    if (!_timestampModeEnabled) {
+      final Pool openFiles = Pool(kMaxOpenFiles);
+      await Future.wait(<Future<void>>[
+        for (File file in files) _hashFile(file, dirty, openFiles)
+      ]);
+    } else {
+      for (File file in files) {
+        _timestampCheckFile(file, dirty);
+      }
+    }
     return dirty;
   }
 
+  void _timestampCheckFile(File file, List<File> dirty) {
+    final String absolutePath = file.path;
+    final String previousTimestamp = previousStamps[absolutePath];
+    // If the file is missing it is assumed to be dirty.
+    if (!file.existsSync()) {
+      currentStamps.remove(absolutePath);
+      previousStamps.remove(absolutePath);
+      dirty.add(file);
+      return;
+    }
+    final String currentTimestamp = file.lastModifiedSync().millisecondsSinceEpoch.toString();
+    if (currentTimestamp != previousTimestamp) {
```

Comment:
It's my understanding that file modification times on Windows aren't high enough resolution for this to work reliably.

Reply:
Yes, that is likely the case.  Since this is an opt-in only flag I think its an okay trade-off